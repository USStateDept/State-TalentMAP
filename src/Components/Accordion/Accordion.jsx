import PropTypes from 'prop-types';

// use role="tablist" for use with aria-multiselectable
// https://www.w3.org/TR/wai-aria-1.1/#aria-multiselectable
const Accordion = ({ children, isMultiselectable, className }) => (
  <ul role="tablist" className={`usa-accordion ${className}`} aria-multiselectable={isMultiselectable}>
    {children}
  </ul>
);

Accordion.propTypes = {
  children: PropTypes.node.isRequired,
  isMultiselectable: PropTypes.bool,
  className: PropTypes.string,
};

Accordion.defaultProps = {
  isMultiselectable: false,
  className: '',
};

export default Accordion;
