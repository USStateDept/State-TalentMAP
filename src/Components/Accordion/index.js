import AccordionItem from './AccordionItem';
import Accordion from './Accordion';

export { default } from './Accordion';
export { Accordion, AccordionItem };
