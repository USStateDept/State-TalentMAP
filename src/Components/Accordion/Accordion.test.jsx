import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Accordion from './Accordion';

describe('AccordionItemComponent', () => {
  const child = (
    <li>
      <button />
      <div className="usa-accordion-content" aria-hidden="false">
        <span />
      </div>
    </li>
  );

  it('receives elements', () => {
    const wrapper = shallow(
      <Accordion>
        {child}
      </Accordion>,
    );
    expect(wrapper.find('ul').props().children).toBe(child);
  });

  it('receives className prop', () => {
    const wrapper = shallow(
      <Accordion className="my-class">
        {child}
      </Accordion>,
    );
    expect(wrapper.find('.my-class').exists()).toBe(true);
  });

  it('receives isMultiselectable prop', () => {
    const wrapper = shallow(
      <Accordion isMultiselectable>
        {child}
      </Accordion>,
    );
    expect(wrapper.find('ul').prop('aria-multiselectable')).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Accordion>
        {child}
      </Accordion>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
