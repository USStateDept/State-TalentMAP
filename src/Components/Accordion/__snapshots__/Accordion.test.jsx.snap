// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItemComponent matches snapshot 1`] = `
<ul
  aria-multiselectable={false}
  className="usa-accordion "
  role="tablist"
>
  <li>
    <button />
    <div
      aria-hidden="false"
      className="usa-accordion-content"
    >
      <span />
    </div>
  </li>
</ul>
`;
